.github/workflows/debian_11.yml

```diff
@@ -0,0 +1,58 @@
+name: debian_11
+
+on: [push, pull_request]
+
+env:
+  CI_MAKE: make -f .gitlab.mk
+
+jobs:
+  debian_11:
+    # We want to run on external PRs, but not on our own internal PRs
+    # as they'll be run by the push to the branch.
+    if: github.event_name == 'push' ||
+        github.event.pull_request.head.repo.full_name != github.repository
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: install packages required for publishing to S3
+        run: |
+          sudo apt-get -y update
+          sudo apt-get install -y procmail createrepo awscli reprepro
```

Comment:
This step is quite long (49 seconds in the last run), but is not needed for most pushes (developer branches) and PRs. Can we refactor the workflow to don't run it when it is not necessary?

Reply:
Right, let's do it right before deploy call.